litestar/_openapi/schema_generation/schema.py

```diff
@@ -274,14 +277,24 @@ def for_field_definition(self, field_definition: FieldDefinition) -> Schema | Re
 
         result: Schema | Reference
 
-        # NOTE: The check for whether the field_definition.annotation is a Pagination type
-        # has to come before the `is_dataclass_check` since the Pagination classes are dataclasses,
-        # but we want to handle them differently from how dataclasses are normally handled.
-        if field_definition.is_optional:
+        if _do_enum_schema(field_definition):
+            annotation = _type_or_first_not_none_inner_type(field_definition)
+            result = create_enum_schema(annotation, include_null=field_definition.is_optional)
```

Comment:
I think I dislike this because we already generate `--` which is `Null` enough, but we add an ugly `null` dropdown option

Reply:
If you look at the request that is sent when you select either `---` or `null` they are different. One excludes the param, one sends an explicit `null`.

Reply:
i think see how this would be needed if the default were not `None` and it was an optional field.

Reply:
`str | None` -> Accepts `String` or `null`, param can't be omitted 
`str | None` with a default of `None` -> Accepts `String` or `null`, param can be omitted.

At least that's how I understand it.

Reply:
We have a matrix of conditions between the annotation of the parameter and its default value. For the annotation, we need to consider whether it is a union with `None`, or not. And we consider whether the param has a default value, or not.

- If annotation is not union with `None`:
    - in any case with default, we don't add `null` to enum members in the schema.
- If annotation is union with `None`:
  - if parameter has default `None`:
      - we wouldn't necessarily _have_ to add the `null` to the enum options in this case as omission is same outcome.
  - if parameter has not `None` default, or parameter has no default:
      - we have to add `null` to the available enums in either of these cases, as it needs to be able to be specified, and isn't default.

> i think see how this would be needed if the default were not None and it was an optional field.

So in the matrix, this is where annotation is union with `None` and has a default `None`, we could omit the `null` value from the list of options. I'm open to being convinced that we should do this, but my preference would be to not do this, for a few reasons:
- inconsistency with the not `None`/no default case
- more complexity, its another condition we have to handle when generating the docs
- I'm not sure that we can police it - i.e., the signature model is going to be configured with `param: MyEnum | None = None` as the field annotation, so if a client sends `?param=null` in their request, the signature model will allow it.

The argument I can think of for allowing a field to be `null` but not allowing it to be provided as `null` is if there should be some distinction between "parameter not provided", and "parameter provided", where `None` is being used as a sentinel value. For this, maybe we could support allowing params to be annotated as union with our `Empty` type, and we could add explicit handling to remove that from openapi spec. One benefit of that is that there is no actual way to specify `Empty` via a query param as it has no encodable value.

Reply:
i think what we have here is the most sensible, im just being weird 😅

Reply:
> im just being weird

this i can relate to